go.mod

```diff
@@ -14,6 +14,7 @@ require (
 	github.com/hashicorp/go-memdb v1.3.1 // indirect
 	github.com/hashicorp/go-uuid v1.0.2
 	github.com/imdario/mergo v0.3.11 // indirect
+	github.com/k0kubun/colorstring v0.0.0-20150214042306-9440f1994b88 // indirect
```

Comment:
This looks like it was added inadvertently by a manual get (the other code changes don't add any modules), hence the failure in https://github.com/Kong/kubernetes-ingress-controller/pull/1268/checks?check_run_id=2468913241

Can you remove it and tidy again?

Reply:
for example
two ingress rule in different namespace
```yaml
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name:  srv1
  namespace: ns1
  annotations:
    kubernetes.io/ingress.class: kong
    cert-manager.io/issuer: "letsencrypt-prod"
spec:
  tls:
    - hosts:
        - foo.com
      secretName: foo
 ---
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name:  srv1
  namespace: ns2
  annotations:
    kubernetes.io/ingress.class: kong
    cert-manager.io/issuer: "letsencrypt-prod"
spec:
  tls:
    - hosts:
        - foo.com
      secretName: foo


```
this will create two certificate in ns1 and ns2,   host foo.com just need one certificate
whether a   certificate  in ns1 or ns2, kong controller lead to  certificate with  host foo.com is not fixed。  

the two ingress rules  is applied wrongly by human, 
 it can solved by remaining only one ingress rule that specified the tls host。

nginx controller can handle this correctly by sorting ingress,  always return a certificate  with namespace/secretName order first。

Reply:
Got it, thanks, that makes sense. We probably should indicate when this happens in logs, but making it not random is an improvement. I've added https://github.com/Kong/kubernetes-ingress-controller/issues/1270 for us to follow up later.